PhysicsTools/PatAlgos/plugins/PATSingleVertexSelector.cc

```diff
@@ -128,7 +128,8 @@ PATSingleVertexSelector::filter_(Mode mode, const edm::Event &iEvent, const edm:
   case FromCand: {
     if (bestCand_.isNull()) return result;
     reco::Vertex vtx;
-    if (typeid(*bestCand_) == typeid(reco::VertexCompositeCandidate)) {
+    auto const& bestCandDeref = *bestCand_;
+    if (typeid(bestCandDeref) == typeid(reco::VertexCompositeCandidate)) {
```

Comment:
for my education, what is this supposed to change? 
just the line at which the code would crash if `bestCand_` is null/invalid?